python/ray/data/grouped_data.py

```diff
@@ -297,13 +313,20 @@ def map_groups(
         else:
             sorted_ds = self._dataset.repartition(1)
 
-        # Returns the group boundaries.
-        def get_key_boundaries(block_accessor: BlockAccessor):
+        def get_key_boundaries(block_accessor: BlockAccessor) -> List[int]:
+            """Compute block boundaries based on the key(s)"""
+
             import numpy as np
 
-            boundaries = []
             # Get the keys of the batch in numpy array format
             keys = block_accessor.to_numpy(self._key)
+
+            if isinstance(keys, dict):
+                # For multiple keys, we generate a separate tuple column
+                convert_to_multi_column_sorted_key = np.vectorize(_MultiColumnSortedKey)
+                keys: np.ndarray = convert_to_multi_column_sorted_key(*keys.values())
```

Comment:
Is this an `ndarray` of `_MultiColumnSortedKey` objects?

Reply:
Yes